ts/sagas/notifications.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * A saga to manage notifications
+ */
+
+import { BasicResponseType } from "italia-ts-commons/lib/requests";
+import { Alert, Platform } from "react-native";
+import { call, Effect, select, takeLatest } from "redux-saga/effects";
+
+import { Installation } from "../../definitions/backend/Installation";
+import { PlatformEnum } from "../../definitions/backend/Platform";
+import { BackendClient } from "../api/backend";
+import { apiUrlPrefix, environment } from "../config";
+import I18n from "../i18n";
+import { SESSION_INITIALIZE_SUCCESS } from "../store/actions/constants";
+import {
+  InstallationState,
+  notificationsInstallationSelector
+} from "../store/reducers/notifications/installation";
+import { sessionTokenSelector } from "../store/reducers/session";
+
+const notificationsPlatform: PlatformEnum = Platform.select({
+  ios: PlatformEnum.apns,
+  android: PlatformEnum.gcm
+});
+
+function notifyNotificationsInstallationError() {
+  // If the application is running in dev mode show an alert
+  if (environment === "DEV") {
+    Alert.alert(
+      I18n.t("notifications.installation.errorTitle"),
+      I18n.t("notifications.installation.errorMessage")
+    );
+  }
+}
+
+/**
+ * This generator function calls the ProxyApi `installation` endpoint
+ */
+function* updateInstallation(): Iterator<Effect> {
+  // Get the notifications installation data from the store
+  const notificationsInstallation: InstallationState = yield select(
+    notificationsInstallationSelector
+  );
+
+  // Get the token from the state
+  const sessionToken: string = yield select(sessionTokenSelector);
```

Comment:
isn't `sessionToken: string | undefined` ? unless the next `if` is useless

Reply:
Yes it is string | undefined. After `yield` we loose types :(